### ISSUE TITLE
Noisy warning - torch.fx.experimental.symbolic_shapes: [WARNING] Ignored guard (...), this could result in accuracy problems

### ISSUE THREAD

**Issue by philgzl** [pytorch/pytorch#101265]
### 🐛 Describe the bug

This warning is raised literally thousands of times when compiling my model. Not sure what it means nor if I should be worried.

I managed to write a smaller code that produces the warning only once:

```python
import torch

CHANNELS = 4


class MyModule(torch.nn.Module):
    def __init__(self):
        super().__init__()
        self.conv = torch.nn.Conv2d(1, 1, 1)

    def forward(self, x, emb):
        x = self.conv(x) + emb  # (batch_size, 1, ...) + (batch_size, CHANNELS, ...)
        return x


net = MyModule()
net = net.cuda()

opt_net = torch.compile(net, dynamic=True)

batch_sizes = [7, 5, 8]
lengths = [297, 120, 361]
for batch_size, length in zip(batch_sizes, lengths):
    print(f'batch_size={batch_size}, length={length}')
    x = torch.randn(batch_size, 1, 256, length, device='cuda')
    emb = torch.randn(batch_size, CHANNELS, 1, 1, device='cuda')
    out = opt_net(x, emb)
    out.mean().backward()
```

Produces the following output:
```
batch_size=7, length=297
batch_size=5, length=120
[2023-05-11 23:06:55,488] torch.fx.experimental.symbolic_shapes: [WARNING] Ignored guard 256*s0*s1 < 2147483648 == True, this could result in accuracy problems
batch_size=8, length=361
```

I note that by doing **any** of the following, the warning is prevented:
* Setting `CHANNELS=1` such that the output from `self.conv` does not need to be expanded along the channel dimension when doing `x = self.conv(x) + emb`
* Skipping/commenting out `out.mean().backward()`
* Setting `batch_size` to be constant
* Setting `length` to be constant

### Versions

Collecting environment information...
PyTorch version: 2.1.0.dev20230508+cu121
Is debug build: False
CUDA used to build PyTorch: 12.1
ROCM used to build PyTorch: N/A

OS: Scientific Linux release 7.9 (Nitrogen) (x86_64)
GCC version: (GCC) 12.3.0
Clang version: Could not collect
CMake version: version 3.26.3
Libc version: glibc-2.17

Python version: 3.10.11 (main, Apr 15 2023, 23:15:15) [GCC 11.3.0] (64-bit runtime)
Python platform: Linux-3.10.0-1160.88.1.el7.x86_64-x86_64-with-glibc2.17
Is CUDA available: True
CUDA runtime version: Could not collect
CUDA_MODULE_LOADING set to: LAZY
GPU models and configuration: 
GPU 0: Tesla V100-PCIE-16GB
GPU 1: Tesla V100-PCIE-16GB

Nvidia driver version: 530.41.03
cuDNN version: Could not collect
HIP runtime version: N/A
MIOpen runtime version: N/A
Is XNNPACK available: True

CPU:
Architecture:          x86_64
CPU op-mode(s):        32-bit, 64-bit
Byte Order:            Little Endian
CPU(s):                24
On-line CPU(s) list:   0-23
Thread(s) per core:    1
Core(s) per socket:    12
Socket(s):             2
NUMA node(s):          2
Vendor ID:             GenuineIntel
CPU family:            6
Model:                 85
Model name:            Intel(R) Xeon(R) Gold 6126 CPU @ 2.60GHz
Stepping:              4
CPU MHz:               999.914
CPU max MHz:           3700.0000
CPU min MHz:           1000.0000
BogoMIPS:              5200.00
Virtualization:        VT-x
L1d cache:             32K
L1i cache:             32K
L2 cache:              1024K
L3 cache:              19712K
NUMA node0 CPU(s):     0-11
NUMA node1 CPU(s):     12-23
Flags:                 fpu vme de pse tsc msr pae mce cx8 apic sep mtrr pge mca cmov pat pse36 clflush dts acpi mmx fxsr sse sse2 ss ht tm pbe syscall nx pdpe1gb rdtscp lm constant_tsc art arch_perfmon pebs bts rep_good nopl xtopology nonstop_tsc aperfmperf eagerfpu pni pclmulqdq dtes64 monitor ds_cpl vmx smx est tm2 ssse3 sdbg fma cx16 xtpr pdcm pcid dca sse4_1 sse4_2 x2apic movbe popcnt tsc_deadline_timer aes xsave avx f16c rdrand lahf_lm abm 3dnowprefetch epb cat_l3 cdp_l3 invpcid_single intel_ppin ssbd mba rsb_ctxsw ibrs ibpb stibp tpr_shadow vnmi flexpriority ept vpid fsgsbase tsc_adjust bmi1 hle avx2 smep bmi2 erms invpcid rtm cqm mpx rdt_a avx512f avx512dq rdseed adx smap clflushopt clwb intel_pt avx512cd avx512bw avx512vl xsaveopt xsavec xgetbv1 cqm_llc cqm_occup_llc cqm_mbm_total cqm_mbm_local dtherm ida arat pln pts hwp hwp_act_window hwp_pkg_req pku ospke md_clear spec_ctrl intel_stibp flush_l1d arch_capabilities

Versions of relevant libraries:
[pip3] flake8==6.0.0
[pip3] numpy==1.24.3
[pip3] pytorch-triton==2.1.0+7d1a95b046
[pip3] torch==2.1.0.dev20230508+cu121
[pip3] torch-ema==0.3
[pip3] torchaudio==2.1.0.dev20230508+cu121
[pip3] torchvision==0.16.0.dev20230508+cu121
[pip3] triton==2.0.0
[conda] Could not collect

cc @ezyang @soumith @msaroufim @wconstab @ngimel @bdhirsh @anijain2305

**Comment by ezyang** [comment]
This is a known problem we plan to fix soon. But maybe we should make it less chatty in the meantime. Whether or not the warning is serious or not depends on the printed guard; if this guard always happens to be satisfied the warning is benign.

**Comment by steelep** [comment]
Would love to see this fixed, until then: 
```
   os.environ["TORCH_LOGS"] = "0"
```
or
```
   os.environ["TORCH_WARN_ONCE"] = "1"
```

**Comment by j93hahn** [comment]
@ezyang have you guys fixed this issue yet? I am seeing a similar thing on my end with PyTorch 2.4

**Comment by umarbutler** [comment]
I can report the same as @j93hahn on 2.4.1.

**Comment by j93hahn** [comment]
@umarbutler I have a fix if you're wrapping a model class with `torch.compile`:

Instead of doing `model = torch.compile(nn.Linear(728, 10))`, try:
```
model = nn.Linear(728, 10)
model = torch.compile(model)
```

I think the symbolic linker likes to be able to know the object's internals before creating the dynamo graph, but that's just my theory
